### DISCUSSION TITLE
Combat, Hero screen, during a combat, when the player accesses the hero's screen, the creatures still present in the hero's army are not updated

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3541] [Ideas]
In fheroes2, during a combat, when the player accesses the hero's screen, the creatures still present in the hero's army are not updated.
It is the starting army that is displayed.

In fheroes2:

![Hero army not updated 001](https://user-images.githubusercontent.com/43583994/119536544-7bcaf580-bd89-11eb-9ebd-331120a46454.png)

![Hero army not updated 002](https://user-images.githubusercontent.com/43583994/119536561-81c0d680-bd89-11eb-9a48-676106f3cdd5.png)

![Hero army not updated 003](https://user-images.githubusercontent.com/43583994/119536574-86858a80-bd89-11eb-9d02-379569eca439.png)

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , yes, we don't update creatures and we show them as before the battle which might be more useful to know how many troops you lost.

**Comment by LeHerosInconnu** [comment]
@ihhub.

> Hi @LeHerosInconnu , yes, we don't update creatures and we show them as before the battle which might be more useful to know how many troops you lost.

That is fine.
I think it would also be interesting to specify this with the current number of creatures and the number of starting creatures for each troop that way (instead of just the number of starting creatures):

"0/5"
"0/32"
"2/15"
"0/3"
"0/6"

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , could you please explain where exactly "0/15" number should be displayed?

**Comment by LeHerosInconnu** [comment]
@ihhub.

> Hi @LeHerosInconnu , could you please explain where exactly "0/15" number should be displayed?

Simply at the place where the number is already displayed, in the lower right corner on the creature graphic.
The current number of creatures is added just before the starting number, per example for Battle Dwarves, "2/" is added before "15", and "0/" is added for other troops completely decimated:

![Hero army not updated 003 modifié 02](https://user-images.githubusercontent.com/43583994/119594123-7b5b4a80-bddb-11eb-9346-dcfe368a6547.png)

**Comment by Branikolog** [comment]
Hi, @ihhub & @LeHerosInconnu !
This seems to look like a nice enhancement for the new option we've recently implemented.
I'm not sure it can easily be implemented so we need to make a separate suggestion issue for this idea.

**Comment by LeHerosInconnu** [comment]
Hello @ihhub and @Branikolog,

Also, it is not possible to access the screen of other heroes with the arrows on each side at the bottom of the screen.
Why this limitation, the player may need to consult his other heroes to, for example, make the decision to run away or surrender with the hero who is currently fighting.
Thus, only actions on creatures should not be possible (separating troops, moving troops, dismissing troops, and that only for the hero currently in combat).
And the player should also be able to move artifacts.

Of course, the added information about the number of current creatures ("0/", "2/") should be displayed only for the screen of the hero who is currently in combat).

> Hi, @ihhub & @LeHerosInconnu !
> This seems to look like a nice enhancement for the new option we've recently implemented.
> I'm not sure it can easily be implemented so we need to make a separate suggestion issue for this idea.

If necessary, I can modify this issue and add another issue for the possibility to access the screens of the other heroes, and one more for the possibility to move the artifacts.

**Comment by oleg-derevenetz** [comment]
Hi @LeHerosInconnu

> And the player should also be able to move artifacts.

What is the purpose of this from practical point of view?

**Comment by LeHerosInconnu** [comment]
Hello @oleg-derevenetz,

> Hi @LeHerosInconnu
> 
> > And the player should also be able to move artifacts.
> 
> What is the purpose of this from practical point of view?

Here is the answer I wrote in another issue: :)

Hello @ihhub,

> Hi @LeHerosInconnu and @Branikolog why do even need this? Is there any logical situation when a player needs to move artifacts or monsters while leveling up?

You have to ask yourself the question in a different way. ;)

Why prevent the user from performing these actions while in this screen?

If the user wants to perform these actions, it's because he has the opportunity to do it now.

(I discovered this flaw precisely because I wanted to take the opportunity to change the location of the artifacts without having to reopen the screen later.)

If the user can't do it, for no good reason in this case, he's going to protest against the GUI because it doesn't work as usual in this screen; and it forces him to reopen the hero's screen to perform the actions he wanted to do.

Therefore, it is preferable to let the user change the position of the artifacts, dismiss his creatures, change the formation of his troops, and so on. :)

_Originally posted by @LeHerosInconnu in https://github.com/ihhub/fheroes2/issues/2004#issuecomment-735819801_

**Comment by oleg-derevenetz** [comment]
Hi @LeHerosInconnu

> he's going to protest against the GUI because it doesn't work as usual in this screen

It is already "doesn't work as usual" because player is not able to move/dismiss troops on this screen (for obvious reasons), so it's obvious that this screen is read only. Not to mention the fact that such ability is not available in original game at all.

**Comment by LeHerosInconnu** [comment]
@oleg-derevenetz.

> It is already "doesn't work as usual" because player is not able to move/dismiss troops on this screen (for obvious reasons), so it's obvious that this screen is read only.

For actions on troops it's easily understandable, it's during a combat and it couldn't (and shouldn't) be taken into account.

> Not to mention the fact that such ability is not available in original game at all.

We are here to improve the original game, making fheroes2 the best designed for a fantastic gaming experience. :D

**Comment by Branikolog** [comment]
@LeHerosInconnu 
To my mind, access to heroes that are not involved in a battle is not such a "musthave" feature and I'm pretty sure there would be really a few cases, where this could be useful... 
I find it quite logical, that I can look through info of a hero, I currently have a battle with, but not others.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> @LeHerosInconnu
> To my mind, access to heroes that are not involved in a battle is not such a "musthave" feature and I'm pretty sure there would be really a few cases, where this could be useful...
> I find it quite logical, that I can look through info of a hero, I currently have a battle with, but not others.

It is when you need it that you realize that it is missing.

This will depend on the different types of players.
Personally, I like to use several "main heroes" during my games.
And it's not always easy to remember how the heroes are set up (secondary skills learned, level, artifacts owned, spells known, etc., etc.) especially on large maps that can be played over several days.

It often happens to me, during combat, that I want to consult the other heroes to know, for example, if the one who is fighting is expendable, or if he is the only one who knows the precious adventure spell that I use every turn or the "Summon Boat" spell. :)

**Comment by ihhub** [comment]
I'm moving this issue into discussion section till we find a proper solution for everyone.

**Reply by ihhub** [reply]
If I understood correctly, @LeHerosInconnu , @Branikolog and @oleg-derevenetz , we agreed to display `0/N` numbers in hero screen. If it is @LeHerosInconnu please create a separate issue just for this :)

**Reply by LeHerosInconnu** [reply]
I think it was okay for "X/N" numbers in hero screen.
It's here: https://github.com/ihhub/fheroes2/issues/3542.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10098, https://github.com/ihhub/fheroes2/discussions/10099.
There is an issue for this: .https://github.com/ihhub/fheroes2/issues/3542.
I close the discussion.
